.github/workflows/ci.yml

```diff
@@ -7,20 +7,57 @@ on:
   pull_request:
 
 jobs:
+  check-integrity:
+    name: Check integrity
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe #v4.1.0
+        with:
+          go-version: 1.19
+
+      - name: Checkout code
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 #v3.5.3
+
+      - name: Check Go modules dependency file integrity
+        run: |
+          for module_file in $(find . -type f -name go.mod); do
+            module=$(dirname $module_file)
```

Comment:
Right now there is no folder name with whitespace but I still RECOMMEND to prepare the script for the worst case.

This snippet handles correctly the whitespace:
```
for module_file in "$(find . -type f -name go.mod)"; do
    module=$(dirname "$module_file")
```

Reply:
I have some problems applying this suggestion, de script doesn't seem to work. I will try to figure it out why but could you please take a look @szykes and validate that this suggestion works properly?

Reply:
Mea culpa. This is a truly working solution:

```
find . -type f -name go.mod | while read module_file ; do
    module=$(dirname "$module_file")
```